feat: Workspace 2.0 new design 🎉

shariquerik opened the pull request:
New Features and Fixes:

<details>

<summary>Implemented Single Card design</summary>

https://user-images.githubusercontent.com/30859809/146336475-6f85d861-105f-494f-9ee1-922a6960ae23.mov

</details>
<details>

<summary>Added a new button on the left side of the block on hover</summary>

https://user-images.githubusercontent.com/30859809/146338631-9a12a7fb-d3a3-4b00-8db2-fe86d509949e.mov

</details>
<details>

<summary>Edit Page Title and other details</summary>

https://user-images.githubusercontent.com/30859809/146337157-2d7dd7ab-e874-4446-8ad0-e3c018aa49b4.mov

</details>
<details>

<summary>Duplicate Page</summary>

https://user-images.githubusercontent.com/30859809/146338659-7e888e45-e640-40b4-b48d-afb4698afe97.mov

</details>
<details>

<summary>Drag to resize blocks</summary>

https://user-images.githubusercontent.com/30859809/146337486-a51b7b65-948c-4524-88a0-79d36976837f.mov

</details>
<details>

<summary>Rearranged and fixed icons in Icon picker</summary>

https://user-images.githubusercontent.com/30859809/146337519-f5cf4005-e3e0-4ad5-8cde-2d81faa61588.mov

</details>
<details>

<summary>Dropdown Menu for blocks instead of tune button</summary>

https://user-images.githubusercontent.com/30859809/146337557-0b6dd3da-8b5a-4699-aca8-3af72076793e.mov

</details>
<details>

<summary>Header size inline tool for the header and text block (remove from tune button)</summary>

https://user-images.githubusercontent.com/30859809/146337593-e11c588b-8d38-45bc-92f3-19f018c52792.mov

</details>

>docs: https://frappeframework.com/docs/v14/user/en/desk/workspace

shariquerik committed:
fix: Added new icons for each wspace blocks

shariquerik committed:
fix: Rearranged and fixed icons

shariquerik committed:
feat: Add new block new design

shariquerik committed:
feat: Drag to resize blocks

shariquerik committed:
fix: Header Size Inline Tool

shariquerik committed:
feat: single card design

shariquerik committed:
fix: removed unused code from container.js

shariquerik committed:
fix: validate_link error while updating shortcut

shariquerik committed:
fix: added title on card links

shariquerik committed:
fix: small fixes

shariquerik committed:
fix: Handled dark mode

shariquerik committed:
feat: Edit page title, icon, etc from sidebar

shariquerik committed:
feat: Duplicate Page from sidebar

shariquerik committed:
fix: minor paragraph block fix

shariquerik committed:
fix: sider fix

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: add new block glitch because of blur event

shariquerik committed:
fix: while duplicating was not updating icon

Alchez commented:
@shariquerik, are there plans to backport these changes to v13 as well?

shariquerik commented:
> @shariquerik, are there plans to backport these changes to v13 as well?

No, Workspace 2.0 will be released in v14, it will not be backported to v13

Komal-Saraf0609 commented:
@shariquerik Some observations:

1. Clicking on "+" button multiple times causes the menu to appear multiple times. Refer below SS:

![Screenshot 2021-12-27 at 1 20 02 PM](https://user-images.githubusercontent.com/81952590/147448514-0b40619e-a1df-421a-9a80-f80d42a5f10e.png)

2. When the onboarding block is resized to its minimum size the UI for it appears to be broken. Refer SS:

![Screenshot 2021-12-27 at 1 23 10 PM](https://user-images.githubusercontent.com/81952590/147448804-f61569c2-d159-47fe-b180-0237b548908f.png)

3. At the time of editing the workspace, all the links already present on the desk should be unclickable. Bcoz what is happening now is whenever I make some customization on the desk and if I click on any of the shortcut and try to come back to the desk, all of my customizations are lost. Refer below recording:

https://user-images.githubusercontent.com/81952590/147449362-0643e7fd-053f-404c-af50-581eff2201b9.mov

4. After creating a new page you cant revisit it using the sidebar. Refer below recording:

https://user-images.githubusercontent.com/81952590/147449698-cecf6e73-b21a-4f6d-9417-e8bcd8e7f06d.mov

5. When creating a card and adding multiple links to it, there is a case when I can add mutiple links there but when I add some links and any one of the link is linked to a a report or say page which does not exist, in that case it gives an error only after I try to save the customizations and all of my other customization in the card will be lost because it doesnt create the card at all. Refer the below recording:

https://user-images.githubusercontent.com/81952590/147450761-9a37b961-89d8-4561-91d8-1ec0e38a846f.mov

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: Workspace Sidebar, Edit, New Page Fixes

Bugs:
- Clicking on + button multiple time opens multiple block list
- Resizing Onboarding block breaks design
- In edit mode links should not be clickable
- While creating new page we cannot revisit it from sidebar
- While adding multiple links in card getting error

Refactor:
- All Sidebar actions happens while you do it, no need to click Save Customization button
- Updating Sidebar, Creating New Page, Duplicating every action is much faster
- Most of the actions are happening in background while rending page using cached data

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: sider fix

shariquerik committed:
fix: set correct seq_id while duplicating page

shariquerik committed:
fix: don't show current page in parent's field while editing page details

shariquerik committed:
Merge branch 'develop' into wspace-new-design

rmehta commented on frappe/public/js/frappe/views/workspace/blocks/paragraph.js:
```diff
@@ -27,62 +27,95 @@ export default class Paragraph extends Block {
+			</div>
+		`);
+
+		let all_blocks = frappe.wspace_block.blocks;
```
@shariquerik please remove `wspace` and replace with `workspace` everywhere. As a rule, we don't use abbreviations or short forms as it creates entropy for new people wanting to contribute also kills the readability of the code.

shariquerik committed:
chore: renamed wspace to workspace

shariquerik committed:
fix: while deleting page save customization button should not be visible

shariquerik committed:
test: Updated workspace test based on new design

shariquerik commented:
Unrelated test failing

shariquerik commented:
I will update the documentation once new pages will be created for v14 on frappeframework.com

Docs: https://frappeframework.com/docs/v14/user/en/desk/workspace

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: added notification for workspace creation, editing, duplicating and rearranging sidebar pages

shariquerik committed:
fix: updated standard workspace json files

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: unescape html while making blocks

shariquerik committed:
fix: sider fix

shariquerik committed:
Merge branch 'develop' into wspace-new-design

Komal-Saraf0609 commented:
@shariquerik Retested the feature. Everything seems working fine. We can go ahead with merging the PR.

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: updated code to make workspace responsive

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
fix: removed is_page_allowed method while getting sidebar items (method is useless)

shariquerik committed:
fix: while adding new block via dialog if dialog is closed empty space stays on the page

shariquerik committed:
fix: don't show last shortcut divider. Also handled responsive behaviour

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
test: fixed failing workspace test

shariquerik committed:
test: fixed failing control icon test

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
Merge branch 'develop' into wspace-new-design

shariquerik committed:
Merge branch 'develop' into wspace-new-design
